astropy/coordinates/sky_coordinate.py

```diff
@@ -172,6 +173,15 @@ def __init__(self, *args, **kwargs):
             raise ValueError('Cannot create a SkyCoord without data')
 
     @property
+    def info(self):
+        """
+        Mixin column information (attributes)
+        """
+        if not hasattr(self, '_info'):
+            self._info = ColumnInfo()
+        return self._info
```

Comment:
That would do it.  If you really wanted to go wild, if `QuantityColumn` is a subclass of `Column` you could do something in the base class like like
```python
ColumnInfoClass = ColumnInfo

@lazyproperty
def info(self):
    return self.ColumnInfoClass()
```

while `QuantityColumn` could set `ColumnInfoClass = QuantityColumnInfo` if need be.

I also agree "info" is vague, but I don't think "col_info" is particularly elucidating since if you're using that attribute you would generally already know that the object you're looking it up on is a column.  I think this is a job for documentation just specifying what goes into the ColumnInfo object.  This could be documented both in the docs for the .info property, and for the ColumnInfo object itself.